airbyte-integrations/connector-templates/destination-java/metadata.yaml.hbs

```diff
@@ -0,0 +1,19 @@
+data:
+  allowedHosts:
+    hosts:
+      - "*" # Please change to the hostname of the source.
```

Comment:
Who would change this value and when? It seems like it might be another thing the user needs to fill out. If it's the case, how can we make sure the user fills it? Will it fail in some way if it's not? Should we add this to [the tutorial](https://docs.airbyte.com/connector-development/config-based/tutorial/getting-started)?

Reply:
I feel like its a bit early in the dev process to ask developers to fill this in with an extra prompt. I think the `*` value is valid but we likely don't want wildcards there for security reasons.
I suggest we :
- Document in the tutorial the purpose of the metadata.yaml file and how to fill the fields (https://github.com/airbytehq/airbyte/issues/25612)
- Add a logic in the  CI metadata validation to fail if `*` is used without a specific comment explaining why the wildcard is used. (https://github.com/airbytehq/airbyte/issues/25613)